src/freenet/clients/http/bookmark/BookmarkManager.java

```diff
@@ -369,19 +367,14 @@ public void storeBookmarks() {
 
 			sfs = toSimpleFieldSet();
 		}
-		FileOutputStream fos = null;
-		try {
-			fos = new FileOutputStream(backupBookmarksFile);
+		try (FileOutputStream fos = new FileOutputStream(backupBookmarksFile)){
+
 			sfs.writeToBigBuffer(fos);
-			fos.close();
-			fos = null;
 			if(!FileUtil.renameTo(backupBookmarksFile, bookmarksFile))
```

Comment:
I'd bet that this breaks on windows. If you haven't closed the file, odds are you can't move it.